bundle/src/main/java/com/adobe/acs/commons/wcm/impl/SiteMapServlet.java

```diff
@@ -139,6 +160,39 @@ protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse r
         }
     }
 
+    private Collection<Resource> getAssetFolders(Page page, ResourceResolver resolver) {
```

Comment:
Just to make sure i understand .. this is getting the list of DAM Asset "folder" paths, and then traversing their sub-trees, collecting all the folders under them, which (the folder list) is used later to collect all the assets?

WDYT about forgoing this collection and just walk the resource tree from this list of root folders?

https://github.com/Adobe-Consulting-Services/acs-aem-samples/blob/master/bundle/src/main/java/com/adobe/acs/samples/resources/SampleResourceVisitor.java

https://docs.adobe.com/content/docs/en/cq/5-6-1/javadoc/index.html?org/apache/sling/api/resource/AbstractResourceVisitor.html